src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -565,6 +565,163 @@ JSON example:
+| PulseSequenceType                 | REQUIRED                                                                                                                        | [string][]                              | General description of the pulse sequence used for the scan. Allowed values are any combination of 2D or 3D acquisition, with the read-out type, separated by an underscore: for example “2D_EPI”, “3D_GRASE”, “3D_spiral”.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
+| EchoTime                          | REQUIRED                                                                                                                        | [number][] or [array][] of [numbers][]  | The echo time for the acquisition, specified in seconds. Specify either one number for the total time-series, or provide an array of numbers with different values of each volume. Corresponds to DICOM Tag 0018, 0081 `Echo Time` (please note that the DICOM term is in milliseconds not seconds).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |
```

Comment:
@effigies Hi Chris, last friday we discussed the option to allow 'array of numbers', with the same length as number of volumes, for EchoTime and FlipAngles. I was wondering how we do this practically? I can easily propose it in the common fields in bep005, is that okay? For validator, should I ask Marco to do the same?

Reply:
Yes, let's change it in **Timing Parameters** and **RF & Contrast**.

I would suggest the text:

> The echo time (TE) for the acquisition, specified in seconds. A single value applies to all volumes, or an array provides a value for each volume. Corresponds to DICOM Tag 0018, 0081 `Echo Time` (please note that the DICOM term is in milliseconds not seconds). 4D files with variable TE should only be used when volume timing is critical for interpretation of the data, as in ASL or variable TE fMRI sequences.

and

> Flip angle (FA) for the acquisition, specified in degrees. A single value applies to all volumes, or an array provides a value for each volume. Corresponds to: DICOM Tag 0018, 1314 `Flip Angle`. 4D files with variable FA should only be used when volume timing is critical for interpretation of the data, as in ASL or variable FA fMRI sequences.

Then in the ASL section, we mark them as REQUIRED for ASL sequences.

Reply:
@effigies Thanks for the suggested texts. I will add those today. 

For Requirement level, I did change it now for MagneticFieldStrenght into 'RECOMENDED, but  REQUIRED for Arterial Spin Labeling', and the same for the new MRAcquisitionType field, in imitation of what was done already for EchoTime and SliceTiming. I can do the same for EchoTime and FlipAngle, but I agree that, if more BEP's do this, the requirement level fields might get long. What is prefered, so I can keep it consistent?

Reply:
I think leaving "REQUIRED for Arterial Spin Labeling" is fine. We can reorganize in the future if things start getting hard to read.

Reply:
I agree! I'll ask Marco to take a look into the validator as well for FlipAngle and EchoTime.